### PR TITLE
Made better Search for Browsing Reports

### PR DESCRIPTION
**Pull request by PauliusLu**
- Divided Reports into Pages
- Made Filtering by ReportState
- Made an option to be able to merge (aggregate) all reports for same post into one report.

### CODE REVIEW

**Review by baikauskaite — CHANGES_REQUESTED** [review]
I have a few remarks.

This "Back to list" leads to the `Admin page`.
Screenshot 2021-11-30 at 09 39 21

When I created more than 6 reports the button "Next page" didn't appear. It appeared only after doing a search. I suggest the button should also appear in the non-filtered page.
Screenshot 2021-11-30 at 09 42 08

Also the "Merge together" toggle is confusing. It requires some kind of explanation, for example, a popover when you hover the "Merge together", like Armintas did with the Karma points.

**Comment by baikauskaite** [comment]
Also it would be really nice if the post wouldn't be shown with all the buttons "Report", "Send message", "Back to list" in the report
144006384-a395ac0a-ca16-4d0a-a8e0-c7f77f7c862d

File: Karma/Controllers/AdminController.cs [hunk 1]
```diff
@@ -342,5 +344,79 @@ private void hideReportedPost(int id)
             post.State = (int)Post.PostState.Hidden;
             post.IsVisible = false;
         }
+
+        [HttpGet]
+        //[ValidateAntiForgeryToken]
```
Comment (ArmintasP) [inline Karma/Controllers/AdminController.cs:349]: [ValidateAntiForgeryToken] shouldn't be commented out (applies to all places where such thing is done).

File: Karma/Controllers/AdminController.cs [hunk 2]
```diff
@@ -342,5 +344,79 @@ private void hideReportedPost(int id)
...
+            var reportsDM = new ReportDataModel
+            {
+                Reports = /*reports,*/ pgdReports.Select(m => m.report).ToList(),
+                page = page,
+                sorting = sorting,
+                aggr = aggr
+            };
+            return View(reportsDM);
+        }
+
```
Comment (ArmintasP) [inline Karma/Controllers/AdminController.cs:420]: Use helper methods to shorten your code.

File: Karma/Controllers/AdminController.cs [hunk 3]
```diff
@@ -342,5 +344,79 @@ private void hideReportedPost(int id)
             post.State = (int)Post.PostState.Hidden;
             post.IsVisible = false;
         }
+
+        [HttpGet]
+        //[ValidateAntiForgeryToken]
+        public async Task<IActionResult> ViewReports(int? page, string sorting, string? aggr)
```
Comment (baikauskaite) [inline Karma/Controllers/AdminController.cs:350]: This still needs to be divided to smaller units, but I'll wait for that in the next PR, like we talked.

Thanks for the changes.

**Review by baikauskaite — APPROVED** [review]
Looks good

File: Karma/Models/ReportDataModel.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+using System.Threading.Tasks;
+
+namespace Karma.Models
+{
+    public class ReportDataModel
+    {
+        public List<Report> Reports { get; set; }
+        public int? page { get; set; }
+        public string sorting { get; set; }
+        public string? aggr { get; set; }
```
Comment (ArmintasP) [inline Karma/Models/ReportDataModel.cs:13]: `string aggr ` can be set to null just like `string? aggr`. What's intended here?

**Review by ArmintasP — COMMENTED** [review]
Well done.

File: Karma/Controllers/AdminController.cs (hunk 3 above)
Comment (PauliusLu) [reply Karma/Controllers/AdminController.cs:350]: Yes, working on it.

File: Karma/Controllers/AdminController.cs (hunk 2 above)
Comment (PauliusLu) [reply Karma/Controllers/AdminController.cs:420]: My next PR will be code refactor and this code is already refactored there.

File: Karma/Models/ReportDataModel.cs (hunk 4 above)
Comment (PauliusLu) [reply Karma/Models/ReportDataModel.cs:13]: Thanks for spotting mistake. Fixed.

**Pull request merged**